.github/workflows/pull-request.yml

```diff
@@ -27,81 +29,81 @@ env:
-      - name: build
-        run: |
-          make html VERBOSE=1 SOF_DOC_BUILD=_build_doxy
-          du -shc _build*/*
+      - name: deploy
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          personal_token: ${{ secrets.ACTIONS_DEPLOY_KEY }}
+          publish_dir: ./html/
+          publish_branch: master
+          external_repository: xiulipan/xiulipan.github.io
 
-  lax:
 
-    name: "PIP_IGNORE_INSTALLED=0 requirements-lax.txt"
+  build-lax:
+
+    name: "build with requirements-lax.txt"
     runs-on: ubuntu-20.04
     # Makefile downgrades the Sphinx warnings, they are not errors any more
     env: {LAX: 1}
     steps:
       - uses: actions/checkout@v2
 
-      - name: apt-get install base dependencies
-        run: sudo apt-get -y install $ubuntu_base_deps
-
-      - name: temporary python3-pil HACK because github is out of date
-        run: sudo apt-get -y install libimagequant0 libwebpdemux2 &&
-           wget http://security.ubuntu.com/ubuntu/ubuntu/pool/main/p/pillow/python3-pil_7.0.0-4ubuntu0.2_amd64.deb &&
-           sudo dpkg -i python3-pil_*.deb
-
-      - name: apt-get instead of pip
-        run: sudo apt-get -y install
-               python3-pip python3-setuptools python3-wheel
-               python3-sphinx python3-breathe python3-docutils
-               python3-sphinx-rtd-theme python3-sphinxcontrib.plantuml
-
-      - name: PATH += .local/bin
-        run: echo "$HOME/.local/bin" >> $GITHUB_PATH
-
-      # should be a no-op
-      - name: 'pip install -r scripts/requirements-lax.txt'
-        run: pip install -r scripts/requirements-lax.txt
-
-      - name: config tweaks
-        run: |
-           # sphinx-build < 1.7 doesn't support "auto" and we're not sure
-           # all default modules are "thread-safe"
-           sed -i -e '/SPHINXBUILD/ s/-j *auto/-j 1/' Makefile
-           # We don't want plantUML to raise the contribution bar
-           sed -i -e 's/^\(plantuml_output_format *=\).*/\1 "none"/' conf.py
-
-      # No YAML anchors; this is copy/paste
-      - name: configure SOF doc
-        run: |
-          git clone https://github.com/thesofproject/sof
-          cmake -GNinja -S sof/doc -B _build_doxy
-
-      - name: build
+      - name: setup env with requirements-lax.txt
         run: |
-          make html VERBOSE=1 SOF_DOC_BUILD=_build_doxy
-          du -shc _build*/*
+          sudo apt-get update
+          sudo apt-get -y install $ubuntu_base_deps
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+          PIP_IGNORE_INSTALLED=0 pip install -r scripts/requirements-lax.txt
+          # config tweaks
+          # sphinx-build < 1.7 doesn't support "auto" and we're not sure
+          # all default modules are "thread-safe"
+          sed -i -e '/SPHINXBUILD/ s/-j *auto/-j 1/' Makefile
+          # We don't want plantUML to raise the contribution bar
+          sed -i -e 's/^\(plantuml_output_format *=\).*/\1 "none"/' conf.py
```

Comment:
Don't reduce the number of steps, this makes more time consuming to see what failed. This is also completely unrelated to the "deploy" topic of this PR, at the very least it should be submitted in a different PR (if at all)

Reply:
I think we do not need to spilt the setup steps if it very simple with apt update/install and pip install.
PS: could you tell why we need this hack for the lax build？

Reply:
> I think we do not need to spilt the setup steps

Let's discuss this in a different PR. Don't sneakily change my design a couple days after it was merged in a totally unrelated PR.

Reply:
I have mentioned this PR did some optimization as I found some issue during my test

Reply:
> PS: could you tell why we need this hack for the lax build？

You already asked and already answered in PR #320

Reply:
Yep, I got how the `requirements-lax.txt` is used for. 
I tried to remove the `config tweaks` during my test and the build speed will be same with `requirements.txt`
So the main purpose here is to try with a faster config here? @deb-intel @lgirdwood do we really need a duplicated test here？

Reply:
> So the main purpose here is to try with a faster config here?

Maybe: not sure what you mean by "faster config". As I wrote in #320, the purpose is explained inside the `requirements-lax.txt` file. You can also look at the PR that added this file.

Reply:
Thanks for the tips. I carefully read through https://github.com/thesofproject/sof-docs/pull/231 and #237 and got the answer here and add my undersetting in comments